tinyagentos/trace_store.py

```
@@ -437,38 +445,42 @@ async def list(
-                        if ev_id and ev_id in seen_ids:
-                            continue
-                        if ev_id:
-                            seen_ids.add(ev_id)
-                        merged.append(ev)
+                    # Open read-only connection for listing, close after use
+                    conn = await aiosqlite.connect(f"file:{_bucket_db_path(self._data_dir, self._slug, bucket)}?mode=ro", uri=True)
+                    try:
+                        clauses = []
+                        params: list[Any] = []
+                        if kind is not None:
+                            clauses.append("kind = ?"); params.append(kind)
```

Ruff (0.16.3)

[error] Multiple statements on one line (semicolon)

(E702)